onebone/signal/hampel_filter.py

```diff
@@ -0,0 +1,111 @@
+"""hampel_filter.
+- Author: Sunjin Kim
+- Contact: sunjin.kim@onepredict.com
+"""
+from typing import Tuple
+
+import numpy as np
+
+
+def hampel_filter(x: np.ndarray, window_size: int, n_sigmas: float = 3) -> Tuple[np.ndarray, list]:
+
+    """
+    A hampel filter removes outliers.
+    Estimate the median and standard deviation of each sample using
+    MAD(Median Absolute Deviation) in the window range set by the user.
+    If the MAD > 3 * sigma condition is satisfied,
+    the value is replaced with the median value.
+
+    .. math::
+        m_{i} = median(x[window_size])
+        \\\\
+        mad_i = median(|x_{i}-k-m_{i}|,...,|x_{i}+k-m_{i}|})
+        \\\\
+        k_mad = k * mad_i
+    Where :math: `m_i` is Local median,
+    :math: `mad_i` is Median absolute deviation.
+           the residuals (deviations) from the data's median.
+    :math: k_mad is the MAD may be used similarly to how one would use the deviation for the average.
+           In order to use the MAD as a consistent estimator
+           for the estimation of the standard deviation `{\sigma}`,one takes `k * mad_i`
+    :math: `k` is a constant scale factor, which depends on the distribution.
+           For normally distributed data `k` is taken to be `k` = 1.4826
+
+
+    Parameters
+    ----------
+    x : numpy.ndarray
+        1d-timeseries data
+    window_size : int,
+        Lenght of the sliding window.
+        Only integer types are available,
+        and the window size must be adjusted according to your data.
+    n_sigma : float, defalut=3
+        Coefficient of standard deviation.
+
+    Returns
+    ----------
+    filtered_series : numpy.ndarray
+        A value from which Outlier or NaN has been removed by the filter.
+    index : list
+        Returns the index corresponding to Outlier.
+
+    References
+    ----------
+    [1] Pearson, Ronald K., et al. "Generalized hampel filters."
+    EURASIP Journal on Advances in Signal Processing 2016.1 (2016): 1-18.
+    DOI: 10.1186/s13634-016-0383-6
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> fs = 100.0
+    >>> t = np.linspace(0, 1, int(fs))
+    >>> y = np.sin(2 * np.pi *10.0 * t)
+    >>> np.put(y,[9, 13, 24, 30, 45, 51,78],4)
+    >>> first_hampel_filter_array = hampel_filter.hampel_filter(y,2)[0]
+    >>> second_hampel_filter_array = hampel_filter.hampel_filter(y,3)[0]
+
+    .. image:: https://bit.ly/3t55Dlc #nopa
+        :width: 600
+
```

Comment:
아래와 같은 형태로 example 만들면 어떨까요.
- import는 굳이 안써도 될것 같습니다.
- example code도 `make format`이 적용된 형태였으면 좋겠습니다. (띄어쓰기 등)
- 신호를 생성하는 코드 -> 이미지 -> filtering 코드 -> filtering 된 이미지. 이와 같은 형태로 작성되면 좋을것 같습니다.

```suggestion
    >>> fs = 100.0
    >>> t = np.linspace(0, 1, int(fs))
    >>> y = np.sin(2 * np.pi * 10.0 * t)
    >>> np.put(y, [9, 13, 24, 30, 45, 51,78], 4)
    <noise signal image>
    
    >>> filtered_signal = hampel_filter.hampel_filter(y, window_size=2)[0]
    <filter 된 image1>
  
    >>> filtered_signal = hampel_filter.hampel_filter(y, window_size=3)[0]
    <filter 된 image2>
    .. image:: https://bit.ly/3t55Dlc #nopa
        :width: 600

```

Reply:
현재 상태
![image](https://user-images.githubusercontent.com/86699249/158720976-c74b4afa-573c-40b9-83a9-322cbe9ab0cb.png)

Reply:
> 현재 상태 ![image](https://user-images.githubusercontent.com/86699249/158720976-c74b4afa-573c-40b9-83a9-322cbe9ab0cb.png)

감사합니다. 말씀주신게 훨씬 이해하기 좋은거같아서 수정하였습니다.